### PR TITLE
[C API] Initial implementation

### PR DESCRIPTION
**Pull request by VivekPanyam**
### Summary:

An initial basic implementation of a C API for Neuropod.

### Test Plan:

Added a test that loads a model, runs inference, and verifies the output. More unit tests and integration tests should be added in the future

### CODE REVIEW

File: source/neuropod/bindings/c/c_api_internal.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+limitations under the License.
+*/
+
+#pragma once
+
+#include "neuropod/neuropod.hh"
+
+struct NP_Neuropod
+{
+    std::unique_ptr<neuropod::Neuropod> model;
```
Comment (vkuzmin-uber) [inline source/neuropod/bindings/c/c_api_internal.h:22]: nit: if unique_ptr is used explicitly, it makes sense to include <memory> explicitly too. The same about shared_ptr

File: source/neuropod/bindings/c/np_tensor_allocator_internal.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2020 UATC, LLC
```
Comment (vkuzmin-uber) [inline source/neuropod/bindings/c/np_tensor_allocator_internal.h:1]: btw, what is UATC, is it Uber ATG abbrevation?

File: source/neuropod/bindings/c/c_api_internal.h (hunk 1 above)
Comment (VivekPanyam) [reply source/neuropod/bindings/c/c_api_internal.h:24]: Fixed

File: source/neuropod/bindings/c/np_tensor_allocator_internal.h (hunk 2 above)
Comment (VivekPanyam) [reply source/neuropod/bindings/c/np_tensor_allocator_internal.h:1]: `UATC, LLC == Uber ATG`

**Pull request merged**